src/org/jitsi/impl/neomedia/MediaStreamTrackImpl.java

```diff
@@ -0,0 +1,128 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.impl.neomedia;
+
+import org.jitsi.service.neomedia.*;
+
+/**
+ * The JVB implementation of a {@link MediaStreamTrack}.
+ *
+ * @author George Politis
+ */
+public class MediaStreamTrackImpl
+    implements MediaStreamTrack
+{
+    /**
+     * The minimum time (in millis) that is required for the media engine to
+     * generate a new key frame.
+     */
+    private static final int MIN_KEY_FRAME_WAIT_MS = 300;
+
+    /**
+     * The {@link RTPEncodingImpl}s that this {@link MediaStreamTrackImpl}
+     * possesses, ordered by their subjective quality from low to high.
+     */
+    private final RTPEncodingImpl[] rtpEncodings;
+
+    /**
+     * The {@link MediaStreamTrackReceiver} that receives this instance.
+     */
+    private final MediaStreamTrackReceiver mediaStreamTrackReceiver;
+
+    /**
+     * The time (in millis) that this instance last saw a keyframe.
+     */
+    private long lastKeyframeMs;
+
+    /**
+     * Ctor.
+     *
+     * @param mediaStreamTrackReceiver The {@link MediaStreamTrackReceiver} that
+     * receives this instance.
+     * @param rtpEncodings The {@link RTPEncoding}s that this instance
+     * possesses.
+     */
+    public MediaStreamTrackImpl(
+        MediaStreamTrackReceiver mediaStreamTrackReceiver,
+        RTPEncodingImpl[] rtpEncodings)
+    {
+        this.rtpEncodings = rtpEncodings;
+        this.mediaStreamTrackReceiver = mediaStreamTrackReceiver;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public RTPEncodingImpl[] getRTPEncodings()
+    {
+        return rtpEncodings;
+    }
+
+    /**
+     * Gets the {@link MediaStreamTrackReceiver} that receives this instance.
+     *
+     * @return The {@link MediaStreamTrackReceiver} that receives this instance.
+     */
+    MediaStreamTrackReceiver getMediaStreamTrackReceiver()
+    {
+        return mediaStreamTrackReceiver;
+    }
+
+    /**
+     * Updates rate statistics for the encodings of the tracks that this
+     * receiver is managing. Detects simulcast stream suspension/resuming.
+     *
+     * @param pkt the {@link RawPacket} that is causing the update of this
+     * instance.
+     *
+     * @param encoding the {@link RTPEncodingImpl} of the {@link RawPacket} that
+     * is passed as an argument.
+     */
+    void update(RawPacket pkt, RTPEncodingImpl encoding)
+    {
+        long nowMs = System.currentTimeMillis();
+
+        // Update the encoding.
+        SourceFrameDesc sourceFrameDesc = encoding.update(pkt, nowMs);
+        if (sourceFrameDesc == null /* no frame was changed */
+            || !sourceFrameDesc.isIndependent() /* frame is dependent */
+            || nowMs - lastKeyframeMs < MIN_KEY_FRAME_WAIT_MS)
```

Comment:
I've added some comments in the code. Let me know if you think I should elaborate more.